iocore/net/UnixNetVConnection.cc

```diff
@@ -1139,8 +1139,8 @@ UnixNetVConnection::mainEvent(int event, Event *e)
       (write.vio.mutex && wlock.get_mutex() != write.vio.mutex.get())) {
 #ifdef INACTIVITY_TIMEOUT
     if (e == active_timeout)
-#endif
       e->schedule_in(HRTIME_MSECONDS(net_retry_delay));
+#endif
     return EVENT_CONT;
```

Comment:
This appears to be the more critical fix.  It looks like a standard reschedule if you don't get the lock.  But it appears that we only do this if the event is an active timeout (if INACTIVITY_TIMEOUT is set).  It seems that we are not currently building with INACTIVITY_TIMEOUT defined (at least I'm not).  So this change would cause the reschedule to never happen.  So if you don't get the lock the first time, you will never process the event.

I'm not following the description in the JIRA.  Is the concern that we are double freeing an event object?